detox/src/realms/DetoxPrimaryContext.js

```diff
@@ -135,7 +135,7 @@ class DetoxPrimaryContext extends DetoxContext {
     process.env.DETOX_CONFIG_SNAPSHOT_PATH = this[_sessionFile];
     this[_lifecycleLogger].trace(`Serialized the session state at: ${this[_sessionFile]}`);
 
-    if (opts.workerId != null) {
+    if (opts.workerId !== null) {
```

Comment:
It was intended to be so from the beginning. This is why you can see, for example, in `detox/local-cli/test.js` the safety measure:

```js
await detox.init({ workerId: null });
```

Thanks to the freshly fixed integration test I found this issue.